test/e2e/extensions/high_availability_test.go

```diff
@@ -64,9 +64,40 @@ func TestGameServerHealthyAfterDeletingPodWhileOneExtensionsDown(t *testing.T) {
 	assert.NoError(t, framework.AgonesClient.AgonesV1().GameServers(defaultNs).Delete(ctx, readyGs.ObjectMeta.Name, metav1.DeleteOptions{})) // nolint: errcheck
 }
 
+func TestGameServerCreationRightAfterDeletingOneExtensionsPod(t *testing.T) {
+	logger := e2eframework.TestLogger(t)
+	ctx := context.Background()
+
+	if !runtime.FeatureEnabled(runtime.FeatureSplitControllerAndExtensions) {
+		t.Skip("Skip test. SplitControllerAndExtensions feature is not enabled")
+	}
+
+	assert.NoError(t, waitForAgonesExtensionsRunning(ctx))
+
+	list, err := getAgoneseExtensionsPods(ctx)
+	logger.Infof("Length of pod list is %v", len(list.Items))
+	for i := range list.Items {
+		logger.Infof("Name of extensions pod %v: %v", i, list.Items[i].ObjectMeta.Name)
+	}
+	require.NoError(t, err, "Could not get list of Extension pods")
+
+	logger.Infof("Removing one of the Extensions Pods: %v", list.Items[1].ObjectMeta.Name)
+	deleteAgonesExtensionsPod(ctx, t, false)
+
+	endTime := time.Now().Add(30 * time.Second)
+	for time.Now().Before(endTime) {
+		gs := framework.DefaultGameServer(defaultNs)
+		logger.Infof("Creating game-server %s...", gs.Name)
+		readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, defaultNs, gs)
+		require.NoError(t, err, "Could not get a GameServer ready")
```

Comment:
can we avoid waiting, too?

Reply:
I believe we could. Just add a separate method that will return the Gameserver right after it sends a request to through agones client.

Reply:
I added a method the method. Please take a look.